grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy

```diff
@@ -88,4 +88,22 @@ class NewConsentGroupController extends AuthenticatedController {
             render([error: e.message] as JSON)
         }
     }
+
+    def delete() {
+        Issue issue = queryService.findByKey(params.consentKey)
+        if(issue != null) {
+            Collection<StorageDocument> documents = queryService.getDocumentsForProject(params.consentKey)
+            documents?.each {
+                storageProviderService.removeStorageDocument(it, getUser()?.displayName)
+            }
+            def links = ConsentCollectionLink.findAllByProjectKey(params.consentKey)
+            if(links != null) persistenceService.deleteCollectionLinks(links)
+            issueService.deleteIssue(params.consentKey)
```

Comment:
Nit: There's a good amount of business logic here. In general, we should keep controllers thin and move business logic to a service class instead. I think all of this logic could be in `IssueService` instead.